spirv_msl.cpp

```diff
@@ -9909,23 +10733,114 @@ string CompilerMSL::builtin_to_glsl(BuiltIn builtin, StorageClass storage)
 	switch (builtin)
 	{
 
+	// Handle HLSL-style 0-based vertex/instance index.
 	// Override GLSL compiler strictness
 	case BuiltInVertexId:
-		return "gl_VertexID";
-	case BuiltInInstanceId:
-		return "gl_InstanceID";
-	case BuiltInVertexIndex:
-		return "gl_VertexIndex";
-	case BuiltInInstanceIndex:
-		return "gl_InstanceIndex";
-	case BuiltInBaseVertex:
-		return "gl_BaseVertex";
-	case BuiltInBaseInstance:
-		return "gl_BaseInstance";
-	case BuiltInDrawIndex:
-		SPIRV_CROSS_THROW("DrawIndex is not supported in MSL.");
-
-	// When used in the entry function, output builtins are qualified with output struct name.
+		ensure_builtin(StorageClassInput, BuiltInVertexId);
```

Comment:
Considered adding a fixup lambda to the entry point which just `-= gl_BaseVertex` the VertexID rather than subtracting every time it's used?